src/version.c

```diff
@@ -63,6 +68,7 @@ int get_latest_version(void)
 	free(path);
 	free(url);
 	free(tmp_version.version);
+	cached_version = ret;
```

Comment:
Looking at this again, I think setting cached_version to a negative value for the error case makes sense here; if the download is retried, then `swupd_curl_get_file` will be called again due to the negative cached_version.